src/librbd/image/RemoveRequest.cc

```diff
@@ -575,6 +577,11 @@ Context *RemoveRequest<I>::handle_mirror_image_remove(int *result) {
     *result = 0;
   }
 
+  if (m_from_trash_purge) {
+    // image was already removed from the directory
```

Comment:
I think I was thrown off by the comment when going through the review from top-to-bottom. The comment says "it's removed from the directory" but it should really say that both the id object and the directory entry have been removed.